### ISSUE TITLE
Ability to set response headers in a hook

### ISSUE THREAD

**Issue by nitedani** [vikejs/vike#1803]
### Description

I'd like the ability to set response headers in a hook.
This would be useful for, for example https://github.com/magne4000/vite-plugin-vercel, so it could allow users to set cache-control headers in their +config.js file, to leverage Vercels stale-while-revalidate edge cache.

**Comment by brillout** [comment]
Agreed. Although I'd name it `headersResponse` (because of [`pageContext.headers`](https://vike.dev/headers) and https://github.com/vikejs/vike/issues/1268).

How much of a priority is it?

**Comment by gnardini** [comment]
(I think) related to this I'd like to be able to set a cookie from a +data.ts file.

**Comment by brillout** [comment]
@gnardini Indeed with `useConfig()` it would be a nice addition.

In the meantime, how about the following?

```js
// +data.js

export default (pageContext) => {
  pageContext.theCookieIWantToadd = '...'
}
```

Then access it at your `renderPage()` integration.

**Comment by BierDav** [comment]
I use `vike-solid` and I want to set response cookies dynamically like in solid start where you can use `(getRequestEvent() as RequestEvent & { response: Response })?.response?.headers` which allows for modification too. It seams that vike dosen't offer something similar

**Comment by BierDav** [comment]
What is the timeframe until we can get this feature, the workaround is quite tedious for such a simple task.

**Comment by brillout** [comment]
Correct me if I'm wrong but I think if you create a component hook then you get a pretty good DX.

Pointers to Solid Start's docs welcome and I'll compare the DX with what we plan for Vike.

There isn't any ETA, but for sure not before we finish our work on Vike's CLI. (I.e. not before the upcoming `0.5.0` release.)
